doc/default.nix

```diff
@@ -99,20 +100,30 @@ in pkgs.stdenv.mkDerivation {
     nixos-render-docs
   ];
 
-  src = ./.;
+  src = fs.toSource {
+    root = ./.;
+    fileset = fs.unions [
+      (fs.fileFilter (file:
+        file.hasExt "md"
+        || file.hasExt "md.in"
+      ) ./.)
+      ./style.css
+      ./anchor-use.js
+      ./anchor.min.js
+      ./manpage-urls.json
+    ];
```

Comment:
I was interested in the difference, so I built the derivation on master and on this PR:

```
$ $ diff -u4 -r /nix/store/w483x0vv260dikx0wzl49kf6zkd5i673-doc /nix/store/j1nq1anc8hc6gai6h7ib7rv9jjcs9aq5-source
Only in /nix/store/w483x0vv260dikx0wzl49kf6zkd5i673-doc: common.nix
Only in /nix/store/w483x0vv260dikx0wzl49kf6zkd5i673-doc: default.nix
Only in /nix/store/w483x0vv260dikx0wzl49kf6zkd5i673-doc: doc-support
Only in /nix/store/w483x0vv260dikx0wzl49kf6zkd5i673-doc/functions: library
Only in /nix/store/w483x0vv260dikx0wzl49kf6zkd5i673-doc: old
Only in /nix/store/w483x0vv260dikx0wzl49kf6zkd5i673-doc: shell.nix
Only in /nix/store/w483x0vv260dikx0wzl49kf6zkd5i673-doc: tests
```

This does make me think that this would be less error-prone to _exclude_ the following file types:

1. `.nix`
2. `.py`
3. `.txt` (but also: consider removing `doc/old/cross.txt`.)

Reply:
I think it's cleaner to explicitly list what's needed, so that it's directly associated with the build instructions further down. Also this way we don't need to worry about new file types accidentally influencing the build.

Reply:
Right, my thought is that it's expected in docs to have css, js, and json. Sort of the converse, where there's unexpected breakage due to needing to allowlist one sort of file (css, js) but not another (md).